Remove Yotta support from the documentation, tests and build scripts

sbutcher-arm opened the pull request:
## Description

Yotta is no longer supported by Mbed TLS, so should be removed. Specifically, the
following changes have been made:

 * references to yotta have been removed from the main readme and build
   instructions
 * the yotta module directory and build script has been removed
 * yotta has been removed from test scripts such as all.sh and check-names.sh
 * yotta has been removed from other files that that referenced it such as the
   doxyfile and the bn_mul.h header
 * yotta specific configurations and references have been removed from config.h

## Status
**READY**

## Requires Backporting
Yes  
Which branch?
This needs to be backported to `mbedtls-2.1` but not `mbedtls-2.7` which may retain support.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported


## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

sbutcher-arm commented:
Removing the 'needs backports' label, as we've decided not to backport this and remove support only in the `development` branch.

Patater commented on yotta/data/example-authcrypt/README.md:
```diff
@@ -1,68 +0,0 @@
-# Authenticated Encryption Example
```
Consider moving the relevant information here into [the README.md in the new examples repo](https://github.com/ARMmbed/mbed-os-example-tls/blob/master/authcrypt/README.md).

andresag01 replied:
I personally think that this information is already in the new example repo, but most of it is in the [main README.md](https://github.com/ARMmbed/mbed-os-example-tls) instead of the example because otherwise the information would just be duplicated.

sbutcher-arm replied:
I think the examples are already sufficiently well documented, as @andresag01 states, this information is in the main example `README.md`.

Patater commented on yotta/data/example-benchmark/README.md:
```diff
@@ -1,100 +0,0 @@
-# mbed TLS Benchmark Example
```
Consider moving the relevant parts here into [the new README.md](https://github.com/ARMmbed/mbed-os-example-tls/blob/master/benchmark/README.md).

andresag01 replied:
Same comment as in the previous README.md. The setup information, etc is already in the new repo, but its in the main readme instead of the example's readme. However, the last three paragraphs regarding benchmarking results are not there, so I agree that it would be good to move those three to the new repo

sbutcher-arm replied:
I think the examples are already sufficiently well documented, as @andresag01 states, this information is in the main example `README.md`.

Patater commented on yotta/data/example-selftest/README.md:
```diff
@@ -1,82 +0,0 @@
-# mbed TLS Selftest Example
```
We don't have any selftest example in [the new examples repo](https://github.com/ARMmbed/mbed-os-example-tls). Do we want to migrate this example?

RonEld replied:
we have in the [standalone](https://github.com/ARMmbed/mbedtls/blob/development/programs/test/selftest.c) and in [Mbed OS](https://github.com/ARMmbed/mbed-os/blob/master/TESTS/mbedtls/selftest/main.cpp), so I don't think there is a need for that

Patater replied:
OK, it moved to the `mbed-os` repo. That explains why I couldn't find it in the new examples repo.

Patater commented on yotta/data/entropy_hardware_poll.c:
```diff
@@ -1,88 +0,0 @@
-/*
- *  Hardware entropy collector for the K64F, using Freescale's RNGA
```
Does Mbed OS 5 have one of these hardware entropy collector thingies?

andresag01 replied:
It does. [Here](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/trng_api.c) is the one for K64F. However, it works differently from the one here because Mbed OS has its own TRNG API.

Patater commented:
There are conflicts in `tests/scripts/all.sh`. Please consider rebasing to resolve conflicts.

sbutcher-arm commented on yotta/data/README.md:
```diff
@@ -1,103 +0,0 @@
-
-While the two editions share the same code base, there are still a number of differences, mainly in configuration and integration. You should keep in mind those differences when reading some articles in our [knowledge base](https://tls.mbed.org/kb), as currently all the articles are about the standalone edition.
-
-* The mbed OS edition has a smaller set of features enabled by default in `config.h`, in order to reduce footprint. While the default configuration of the standalone edition puts more emphasize on maintaining interoperability with old peers, the mbed OS edition only enables the most modern ciphers and the latest version of (D)TLS.
```
I've created a [Mbed OS PR #8259](https://github.com/ARMmbed/mbed-os/pull/8259) to move this information into the Mbed OS repo.

sbutcher-arm commented:
I've rebased this, added a ChangeLog entry and fixed a couple of mistakes in the original PR. One was the removal of a `rm` command in `all.sh` which did other things from deleting yotta files, and another was the accidental inclusion of a swap file.

@andresag01 / @Patater / @RonEld - Can you please re-review this? I think it's ready to go.

sbutcher-arm commented:
retest
